Drawing/ColorPicker.js

```diff
@@ -240,24 +218,30 @@ const ColorPickerBase = kind({
 		publicClassNames: true
 	},
 
-	render: ({color, css, disabled, onTogglePopup, popupOpen, renderComponent, text}) => {
+	render: ({color, css, disabled, handleClosePopup, handleOpenPopup, popupOpen, renderComponent, text}) => {
 		return (
 			<Row className={css.colorPicker}>
 				<BodyText className={css.colorBodyText} disabled={disabled}>{text}</BodyText>
-				<ContextualButtonBase
+				<SpottableButton
 					className={css.coloredButton}
-					disabled
+					disabled={disabled}
 					minWidth={false}
-					onClick={() => {
-						if (!disabled) {
-							onTogglePopup();
-						}
-					}}
-					open={popupOpen}
-					popupComponent={() => renderComponent}
+					onClick={handleOpenPopup}
 					style={{backgroundColor: color}}
 					type="color"
 				/>
+				<Popup
+					onClose={handleClosePopup}
+					open={popupOpen}
+					position="left"
+					scrimType="transparent"
+				>
+					<Row>
+						<BodyText>{text}</BodyText>
+						<Button className={css.closeButton} icon={'x'} onClick={handleClosePopup} />
```

Comment:
You can replace icon 'x' with sandstone icon code 'closex'.

Reply:
Done